models/image_classification_mobilenet/mobilenet.cpp

```diff
@@ -0,0 +1,1126 @@
+"{ input i          | | Path to input image or video file. Skip this argument to capture frames from a camera.}"
+"{ initial_width    | 0 | Preprocess input image by initial resizing to a specific width.}"
+"{ initial_height   | 0 | Preprocess input image by initial resizing to a specific height.}"
+"{ rgb              | true | swap R and B plane.}"
+"{ crop             | false | Preprocess input image by center cropping.}"
+"{ backend bt         | 0 | Choose one of computation backends: "
+"0: (default) OpenCV implementation + CPU,"
+"1: CUDA + GPU (CUDA),"
+"2: CUDA + GPU (CUDA FP16),"
+"3: TIM-VX + NPU,"
+"4: CANN + NPU}";
+
+using namespace std;
+using namespace cv;
+using namespace dnn;
+
+vector<string> getLabelImagenet1k();
+
+int main(int argc, char** argv)
+{
+    CommandLineParser parser(argc, argv, keys);
+
+    parser.about("Use this script to run classification deep learning networks in opencv Zoo  using OpenCV.");
+    if (argc == 1 || parser.has("help"))
+    {
+        parser.printMessage();
+        return 0;
+    }
+
+    Scalar mean(123.675, 116.28, 103.53);
+    Scalar std(255*0.229, 255*0.224, 255*0.225);
+    double scale = 1.0;
+    Size inpSize(224, 224);
+    int rszWidth = parser.get<int>("initial_width");
+    int rszHeight = parser.get<int>("initial_height");
+    bool swapRB = parser.get<bool>("rgb");
+    bool crop = parser.get<bool>("crop");
+    String model = samples::findFile(parser.get<String>("model"));
+    int backendTargetid = parser.get<int>("backend");
+ 
+    if (model.empty())
+    {
+        CV_Error(Error::StsError, "Model file " + model + " not found");
+    }
+    vector<string> labels =  getLabelImagenet1k();
+
+    Net net = readNet(model);
+    net.setPreferableBackend(backendTargetPairs[backendTargetid].first);
+    net.setPreferableTarget(backendTargetPairs[backendTargetid].second);
+    //! [Open a video file or an image file or a camera stream]
+    VideoCapture cap;
+    if (parser.has("input"))
+        cap.open(parser.get<String>("input"));
+    else
+        cap.open(0, CAP_DSHOW);
+    if (!cap.isOpened())
+        CV_Error(Error::StsError, "Cannot opend video or file");
+    Mat frame, blob;
+    static const std::string kWinName = model;
+    while (waitKey(1) < 0)
+    {
+        cap >> frame;
+        if (frame.empty())
+        {
+            cout << "Frame is empty" << endl;
+            waitKey();
+            break;
+        }
+
+        if (rszWidth != 0 && rszHeight != 0)
+        {
+            resize(frame, frame, Size(rszWidth, rszHeight));
+        }
+
+        //! [Create a 4D blob from a frame]
+        blobFromImage(frame, blob, scale, inpSize, mean, swapRB, crop);
+
+        // Check std values.
+        if (std.val[0] != 0.0 && std.val[1] != 0.0 && std.val[2] != 0.0)
+        {
+            // Divide blob by std.
+            divide(blob, std, blob);
+        }
```

Comment:
One suggestion:

In the upcoming OpenCV 4.8.0 release, there is a new API `blobFromImageWithParams` which applies mean and std when it loads from image. Here is the doc: https://docs.opencv.org/4.x/d6/d0f/group__dnn.html#ga8027ad777236a7ae543c6e4ffa4d544c.

Can you try with this API? You need to build the latest OpenCV though.

Reply:
Yes but there is this [post ]([url]( https://forum.opencv.org/t/image2blobparams-c-and-python/13437))with associated [issue](https://github.com/opencv/opencv/issues/23764)

Reply:
Since we are making c++ demo, I think Python stuff is not a problem for us.